packages/tenderly-hardhat/src/tasks/verify.ts

```diff
@@ -1,34 +1,30 @@
 import { task } from "hardhat/config";
 import { HardhatPluginError } from "hardhat/plugins";
-import { TenderlyService } from "tenderly";
 
+import { HardhatRuntimeEnvironment } from "hardhat/types";
 import { logger } from "../utils/logger";
 import { PLUGIN_NAME } from "../constants";
-import { newCompilerConfig } from "../utils/util";
-import { extractContractData } from "./common";
+import { ContractByName } from "../tenderly/types";
 
-const tenderlyService = new TenderlyService(PLUGIN_NAME);
-
-task("tenderly:verify", "Verifies contracts on Tenderly")
+task("tenderly:verify", "Verifies contracts on Tenderly based on the configuration in hardhat.config.js.")
   .addOptionalVariadicPositionalParam(
     "contracts",
     "Addresses and names of contracts that will be verified formatted ContractName=Address"
   )
   .setAction(verifyContract);
 
-async function verifyContract({ contracts }: any, { config, hardhatArguments, run }: any) {
-  logger.info("Public verification hardhat task has been invoked.");
+async function verifyContract({ contracts }: any, hre: HardhatRuntimeEnvironment) {
+  logger.info("Verification via tenderly:push hardhat task is invoked.");
+
   if (contracts === undefined) {
-    throw new HardhatPluginError(
-      PLUGIN_NAME,
-      `At least one contract must be provided (ContractName=Address). Run --help for information.`
-    );
+    throw new HardhatPluginError(PLUGIN_NAME, `At least one contract must be provided (ContractName=Address)`);
```

Comment:
I am quite sure that this didn't work with libraries before the changes in this PR. But fair point, what are we going to do with libraries? 
These are our tasks, so we can have our own flag that specifies libraries. Something like:
```bash
npx hardhat tenderly:verify Token=0x... --network sepolia --libraries Token={LibraryName: 0x...},{LibraryName2: 0x...}
```
Since this can be too much to handle for user giving he has a lot of contracts to verify. We can import libraries from a file. The file could have all the libraries the user is dependent on (the user needs to specify it), and we could pick only the libraries that are needed for that verification. 
e.g. of libraries.json file structure (same as the solidity compiler libraries structure):
```json
{
   "ContractName": {
      "LibraryName1": "0x...",
      "LibraryName2": "0x...",
   },
   "ContractName2": {
      "LibraryName3": "0x...",
      "LibraryName4": "0x...",
   }
}
```
Then we could see what contracts the user wants to verify and pick the libraries needed for that contract's verification. We could do this fairly easy like:
```typescript
hre.tenderly.verify({
   name: contractName,
   address: contractAddress,
   libraries: getLibrariesForContract(librariesFileName, contractName)
});
```

Reply:
I like this approach. Nice 😁

Reply:
(also, libraries definitely didn't work for tasks but they did for automatic verification, so just want to ensure that it has total feature parity)

Reply:
The hardhat implementation doesn't allow two variadic parameters to be present. Since the first variadic parameter is `ContractName=Address`, we cannot put libraries as the second variadic parameter.

I built a solution that accepts libraries as:
```bash
--libraries ContractName1=LibraryName1:Address1,LibraryName2:Address2#ContractName2=LibraryName3:Address3,LibraryName4:Address4
```
So the `#` is the separator between the contracts.

How does that sound?